src/pkg/services/m365/api/item_pager.go

```diff
@@ -27,6 +27,57 @@ type DeltaUpdate struct {
 	Reset bool
 }
 
+type NextPager[T any] interface {
+	NextPage() (items []T, reset, done bool)
+}
+
+type nextPage[T any] struct {
+	items []T
+	// reset is only true on the iteration where the delta pager's Reset()
+	// is called.  Callers can use it to reset any data aggregation they
+	// currently use.  After that loop, it will be false again, though the
+	// DeltaUpdate will still contain the expected value.
```

Comment:
nit: could you please move this comment to be on `NextPage()` since that's what callers will actually use?

This comment also feels a bit underspecified. What is the behavior if `len(items) > 0 && reset`? Are those items from the page that should be discarded or from the page that should be kept? Or are we defining an API where it's considered impossible to get that result? (see golang docs for `io.Read` for a [similar situation](https://pkg.go.dev/io#Reader)). Whatever behavior we choose, we should at least document it in comments so callers know what to expect and how to handle edge cases

Reply:
Documented, thanks for the call out.

Fwiw i'm not _implementing_ an items>0 && reset == true possibility.  But the interface doesn't enforce it, so I've added expectations there.

Reply:
>Fwiw i'm not implementing an items>0 && reset == true possibility

It's fine if it's not implemented. There is already other code that _assumes_ a specific implementation though (flagged in another comment) so that assumed behavior should be documented